input/vbb/drawStations.js

```diff
@@ -0,0 +1,47 @@
+import { map, switchMap } from "rxjs/operators";
+import { combineLatest, from } from "rxjs";
+import { upsertStation } from "../../output/station";
+import { poll } from "../../rx";
+import { getSavedStations } from './stations/memoryApi'
+
+import {
+  fetchDepartures,
+  departuresToAvgDelay,
+  fetchStopInfo
+} from "./index";
+import { makeIndirectFetch } from "../../fn-helpers";
+
+
+const pollDepartures = stationId =>
+poll(2000, makeIndirectFetch(fetchDepartures)({ stationId }));
+
+
+export const drawStations = (app) => {
+  if (window.stops) { 
+    window.stops.unsubscribe();
```

Comment:
If you want this to work you should update the code on line 41 from `forEach` to `map` and here you should do `window.stops.forEach(stop => stop.unsubscribe())`
I'd also recommend naming the `stops` either `stops$` or `stopsObservable` so it's a bit clearer what is in there.

Overall though I don't think the observable code should end up in `drawStations`.
Main point behind this being the observables part is mostly input, the drawing part is output.